archive_updater_test.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import subprocess
+import os
+import tempfile
+import shutil
+from archive_updater import fill_message_directory
+from unittest import mock
+
+class ArchiveUpdaterFillMessageDirectoryTest(unittest.TestCase):
+        
+    def setUp(self):
+        self.tmp_dir = tempfile.mkdtemp()
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
+    
+    @mock.patch.object(subprocess, 'check_call')
+    def test_update_failure(self, mock_check_call):
+        mock_check_call.side_effect = subprocess.CalledProcessError('Failed to fetch', 'git fetch')
+        self.assertRaises(subprocess.CalledProcessError,
+                          fill_message_directory, '../linux-kselftest/git/0.git',
+                                                 '../lkml-gerrit-bridge/test_data',
+                                                 '5227e860278407173cbfdab594676764153c27e1')
+        
+    @mock.patch.object(subprocess, 'Popen')
+    @mock.patch.object(subprocess, 'call')
+    @mock.patch.object(subprocess, 'check_call')
+    def test_fill_message_directory_success(self, mock_check_call, mock_call, mock_popen):
+        def writeFile(args, stdout):
+            stdout.write('called with: ' + ' '.join(args))
+        mock_check_call.return_value = ''
+        mock_call.side_effect = writeFile
+        mock_popen.return_value.communicate.return_value = [b'success1 success2 success3', '']
+        
+        last_used_hash = fill_message_directory('abc', self.tmp_dir, '')
```

Comment:
nit: I'd say `'path'` or `'archive_path'` here

Makes it easier to understand the expectation below, e.g. out of context one might ask:
Why do we want `git show -C abc`?
`git show -C archive_path` is a bit more self evident.